src/simulation/LoadGenerator.cpp

```diff
@@ -1701,6 +1814,40 @@ LoadGenerator::pretendTransaction(uint32_t numAccounts, uint32_t offset,
                                                          maxGeneratedFeeRate));
 }
 
+std::pair<LoadGenerator::TestAccountPtr, TransactionFramePtr>
+LoadGenerator::createMixedClassicSorobanTransaction(
+    uint32_t ledgerNum, uint64_t sourceAccountId,
+    GeneratedLoadConfig const& cfg)
+{
+    auto const& blend_cfg = cfg.getMixClassicSorobanConfig();
+    uint32_t weights = blend_cfg.payWeight + blend_cfg.sorobanUploadWeight +
+                       blend_cfg.sorobanInvokeWeight;
+    uint32_t roll = rand_uniform<uint32_t>(1, weights);
```

Comment:
an idea for simplification: use `std::discrete_distribution` with mode weights, which will return a mode (index) to use, so we won't need to manually check the values.

Reply:
Good idea! Done in 6647f00a58dc3aa3f46ec35ed342a1c94bc0544f